packages/store/ts/config/v2/store.ts

```diff
@@ -34,7 +49,20 @@ export function resolveStoreTablesConfig<input, scope extends AbiTypeScope = Abi
   }
 
   return Object.fromEntries(
-    Object.entries(input).map(([key, table]) => [key, resolveTableConfig(table, scope)]),
+    Object.entries(input).map(([key, table]) => {
+      const fullInput = isTableShorthandInput(table, scope)
+        ? resolveTableShorthand(table as validateTableShorthand<typeof table, scope>, scope)
+        : table;
+
+      return [
+        key,
+        resolveTableConfig(
+          // if a tableId exists on the input, keep that tableId
+          { tableId: resourceToHex({ type: "table", namespace: "", name: key }), ...fullInput },
```

Comment:
what about undefined? I think that's technically allowed in the optional type?


```suggestion
          { ...fullInput, tableId: fullInput.tableId ?? resourceToHex({ type: "table", namespace: "", name: key }) },
```

Reply:
good catch!